microscopium/serve.py

```diff
@@ -25,7 +25,7 @@
 
 def dataframe_from_file(filename):
     """Read in pandas dataframe from filename."""
-    df = pd.read_csv(filename, index_col=0).set_index('index')
+    df = pd.read_csv(filename, index_col=0).reset_index().set_index('index')
```

Comment:
Not sure I understand why this change is here, what's it fixing?

Reply:
Oh! Good point. Basically, if your dataframe doesn't have a silly column full of numbers (just the column "index" as the left-most column), then it gets put as the index by default, and doesn't have a column "index" to set as the index. Confusing enough? =P So we pop the index and then set "index" as the index. In the new case, it's a no-op, and in the old case, you end up with a useless column of integers. =\ But I think the new case is the correct way to do it. Technically we could just remove everything after `index_col=0`. Then I have to change the test data, though.

Reply:
I should have said: what do you think? Should I change this?

Reply:
No, the thing is that in a properly-formatted table, the index is correct, and we don't want to drop it. With this syntax,

- a proper index (first column, correct name) gets interpreted correctly, if slightly inefficiently
- a crappy index (unnamed numeric column, as in the demo data) gets kept as a useless column

Perhaps I should modify the demo data and just always assume that column 0 is the correct index?

Reply:
I think I see what you're saying. Well, we generate that csv in microscopium, don't we? So yes, we should ensure we write it properly (and modify the demo data to match) and save ourselves a lot of hassle.